### DISCUSSION TITLE
Can I run wezterm.action via wezterm cli, I mean use wezterm cli to triggle wezterm keyassigment action?

### DISCUSSION THREAD

**Discussion by uaisu** [wezterm/wezterm#4232] [Q&A] [answered]
Thank you for creating such an amazing terminal. 

I want a way to select different pane inside wezterm, I find [PaneSelece](https://wezfurlong.org/wezterm/config/lua/keyassignment/PaneSelect.html), It's really cool. but there's a problem for me. 

I want to execute PaneSelect in WezTerm through its CLI interface. I know I can bind it to a key, but it would be even better if I can   integrate this command  into[ Kakoune](https://github.com/mawww/kakoune) internally. Kakoune already include a [script](https://github.com/mawww/kakoune/blob/master/rc/windowing/wezterm.kak) to intergrate Wezterm's split-pane command. 

Does there a way, I can use wezterm cli to triggle wezterm's keybind action. If I can do this, it's really flexible. This way, I can use the command line instead of relying solely on keybindings, which provides more flexibility and, of course, many other benefits.

**Comment by nekowinston** [comment]
It sounds like you're looking for [`wezterm cli activate-pane-direction`](https://wezfurlong.org/wezterm/cli/cli/activate-pane-direction.html) for this functionality.

**Reply by uaisu** [reply]
Thanks for point out this, but i'm not want a particular solution to cycle pane(and the switch pane by direction way is not so cool like PaneSelect). If I want to cycle different pane, this will work. But what I actually want is a way, I can use `wezterm cli` to call some function just like you typed some key and the key can call some function. 

For example, I can bind Ctrl-8 to PaneSelect,then if I type Ctrl-8, I know, under the hood, there must be some function be called to perform the PaneSelect and make me choose a pane. So I want is I can call those function directly without through key-bind. 
 
If I can do this, not just PaneSelect, I can use `wezterm cli` to do everything wezterm's keybind can do. And it's really powerful.

**Comment by wez** [comment, marked as answer]
No, you cannot directly control the GUI via the CLI. The CLI uses the mux protocol to talk to a mux server (including the one built-in to the GUI). The mux server only understands mux related operations. GUI functions like overlays are not part of the mux model and cannot be directly addressed by the mux protocol.

Take a look at https://wezfurlong.org/wezterm/recipes/passing-data.html for information about ways you can pass data from an application running in a pane to lua by triggering an event.

If you search the GH issues/discussions, I think you can find a more detailed example of someone building a dispatcher using those techniques.

**Reply by uaisu** [reply]
Thank you, it's helpful.
